services/execution/enclave/src/file_handler.rs

```diff
@@ -15,26 +15,41 @@
 // specific language governing permissions and limitations
 // under the License.
 
-use anyhow::ensure;
-use anyhow::Result;
-use sgx_types::error::SgxStatus;
+use anyhow::{ensure, Result};
 use teaclave_types::FileAgentRequest;
 
-extern "C" {
-    fn ocall_handle_file_request(p_retval: *mut u32, in_buf: *const u8, in_len: u32) -> SgxStatus;
-}
+cfg_if::cfg_if! {
+    if #[cfg(feature = "mesalock_sgx")]  {
+        use sgx_types::error::SgxStatus;
+
+        extern "C" {
+            fn ocall_handle_file_request(p_retval: *mut u32, in_buf: *const u8, in_len: u32) -> SgxStatus;
+        }
+
+        pub(crate) fn handle_file_request(request: FileAgentRequest) -> Result<()> {
+            let mut rt: u32 = 2;
+            let bytes = serde_json::to_vec(&request)?;
+            let buf_len = bytes.len();
+            let res =
+                unsafe { ocall_handle_file_request(&mut rt as _, bytes.as_ptr() as _, buf_len as u32) };
+            ensure!(res == SgxStatus::Success, "ocall sgx_error = {:?}", res);
+            ensure!(rt == 0, "ocall error = {:?}", rt);
+            Ok(())
+        }
 
-#[allow(dead_code)]
-pub(crate) fn handle_file_request(request: FileAgentRequest) -> Result<()> {
-    let mut rt: u32 = 2;
-    let bytes = serde_json::to_vec(&request)?;
-    let buf_len = bytes.len();
-    let res =
-        unsafe { ocall_handle_file_request(&mut rt as _, bytes.as_ptr() as _, buf_len as u32) };
+    }else{
 
-    ensure!(res == SgxStatus::Success, "ocall sgx_error = {:?}", res);
-    ensure!(rt == 0, "ocall error = {:?}", rt);
-    Ok(())
+        use teaclave_file_agent::ocall_handle_file_request;
```

Comment:
`ocall_handle_file_request` is not suitable for an application running in libos.